llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -369,12 +403,28 @@ buildFunctionsToAspectsMap(Module &M, TypeToAspectsMapTy &TypesWithAspects) {
   }
 
   SmallPtrSet<const Function *, 16> Visited;
-  for (Function *F : EntryPoints)
+  for (Function *F : EntryPoints) {
+    setSyclFixedTargetsMD(F, Targets);
     propagateAspectsThroughCG(F, CG, FunctionToAspects, Visited);
+  }
 
   return FunctionToAspects;
 }
 
+SPAUOptions parseOpts(std::string Params) {
```

Comment:
We should probably accept the string passed to via `StringRef` and then use [`StringRef::split`](https://llvm.org/doxygen/classllvm_1_1StringRef.html#a6414735714bdbd5c8790f32a0fe99bd2)

Reply:
Done [3c69169](https://github.com/intel/llvm/pull/7115/commits/3c691692a7c5e159f9a06af42f32c28c3c729f90)